File: prototype/rails/src/app/domain/shop/status_label.rb
module Domain
  module Shop
    module StatusLabel
      module_function

      # Order and fulfillment states are stored snake_case; a page reads one
      # back as a sentence.
      def humanize(status)
        status.to_s.tr("_", " ").capitalize
      end
    end
  end
end

File: prototype/php/src/app/Domain/Shop/StatusLabel.php
<?php

declare(strict_types=1);

namespace App\Domain\Shop;

final class StatusLabel
{
    /**
     * Order and fulfillment states are stored as snake_case; a page reads them
     * back as a sentence.
     */
    public static function humanize(string $status): string
    {
        return ucfirst(str_replace('_', ' ', $status));
    }
}
